### PR TITLE
Update error message while adding a repo with an out of bound index

### PR DESCRIPTION
**Pull request by takirala**
- Updated error message while adding a repository with an index that would result in out of bound exception
- Added a test case to check the same

### CODE REVIEW

File: cosmos-common/src/main/scala/com/mesosphere/cosmos/error/RepositoryAddIndexOutOfBounds.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ import io.circe.generic.semiauto.deriveEncoder
 
 final case class RepositoryAddIndexOutOfBounds(attempted: Int, max: Int) extends CosmosError {
   override def data: Option[JsonObject] = CosmosError.deriveData(this)
-  override def message: String = s"Index out of range: $attempted"
+  override def message: String = s"Index ($attempted) is out of range. Index value must be between 0 and $max inclusive."
```
Comment (takirala) [inline cosmos-common/src/main/scala/com/mesosphere/cosmos/error/RepositoryAddIndexOutOfBounds.scala]: @larioj We were talking yesterday what if the the `max = 0` but that case never happens due to the check here - https://github.com/dcos/cosmos/blob/master/cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/ZkRepositoryList.scala#L180

Comment (larioj) [inline cosmos-common/src/main/scala/com/mesosphere/cosmos/error/RepositoryAddIndexOutOfBounds.scala]: What will the message be when list.size = 0 and I attempt to add at index 20?

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageRepositorySpec.scala [hunk 2]
```diff
@@ -89,6 +89,18 @@ class PackageRepositorySpec
       assertResult(Status.BadRequest)(status)
       info(expected.message)
     }
+    scenario("the user should be able to add a repository at the end of the list") {
```
Comment (larioj) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageRepositorySpec.scala]: "the user should be able to add a repository at the end of the list using an index"

File: cosmos-common/src/main/scala/com/mesosphere/cosmos/error/RepositoryAddIndexOutOfBounds.scala [hunk 3]
```diff
@@ -6,7 +6,10 @@ import io.circe.generic.semiauto.deriveEncoder
 
 final case class RepositoryAddIndexOutOfBounds(attempted: Int, max: Int) extends CosmosError {
   override def data: Option[JsonObject] = CosmosError.deriveData(this)
-  override def message: String = s"Index out of range: $attempted"
+  override def message: String = s"Index ($attempted) is out of range. Index value " + s"${
+      if (max == 0) "can be at most 0"
```
Comment (larioj) [inline cosmos-common/src/main/scala/com/mesosphere/cosmos/error/RepositoryAddIndexOutOfBounds.scala]: What do you think of 'must be 0'?

Comment (jsancio) [inline cosmos-common/src/main/scala/com/mesosphere/cosmos/error/RepositoryAddIndexOutOfBounds.scala]: Too fancy in my opinion:

```scala
override def message: String = {
  val suggestion = if (max == 0) "can be at most 0" else s"must be between 0 and $max inclusive."
  s"Index ($attempted) is out of range. Index value " + suggestion
}
```

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageRepositorySpec.scala [hunk 4]
```diff
@@ -89,6 +89,18 @@ class PackageRepositorySpec
...
+      val name = "bounds"
+      val index = defaultRepositories.size
+      val repository = rpc.v1.model.PackageRepository(name, uri)
+      val (status, actual) = ItUtil.addRepositoryEither(repository, Some(index))
+      val expected = rpc.v1.model.PackageRepositoryAddResponse(
+        originalRepositories :+ repository
+      )
+      assertResult(Right(expected))(actual)
+      assertResult(Status.Ok)(status)
+    }
```
Comment (jsancio) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageRepositorySpec.scala]: We should add newline before and after the `scenario`. I see that the rest of the code doesn't do this. You can leave it as is or fix the rest of the code.

**Pull request merged**